frontend/server/src/main/java/com/amazonaws/ml/mms/util/codec/MessageCodec.java

```diff
@@ -0,0 +1,159 @@
+        out.writeInt(modelInputs.getName().length());
+        out.writeBytes(modelInputs.getName().getBytes(Charset.forName("UTF-8")));
+
+        if (modelInputs.getContentType() != null) {
+            out.writeInt(modelInputs.getContentType().length());
+            out.writeBytes(modelInputs.getContentType().getBytes(Charset.forName("UTF-8")));
+        } else {
+            out.writeInt(0); // Length 0
+        }
+
+        out.writeInt(modelInputs.getValue().length);
+        out.writeBytes(modelInputs.getValue());
+    }
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, BaseModelRequest msg, ByteBuf out) {
+        if (msg instanceof ModelLoadModelRequest) {
+            out.writeDouble(1.0); // SOM
+            out.writeInt(1); // load 1
+            out.writeInt(msg.getModelName().length());
+            out.writeBytes(msg.getModelName().getBytes(Charset.forName("UTF-8")));
+
+            out.writeInt(((ModelLoadModelRequest) msg).getModelPath().length());
+            out.writeBytes(
+                    ((ModelLoadModelRequest) msg)
+                            .getModelPath()
+                            .getBytes(Charset.forName("UTF-8")));
+
+            if (((ModelLoadModelRequest) msg).getBatchSize() >= 0) {
+                out.writeInt(((ModelLoadModelRequest) msg).getBatchSize());
+            } else {
+                out.writeInt(1);
+            }
+
+            out.writeInt(((ModelLoadModelRequest) msg).getHandler().length());
+            out.writeBytes(
+                    ((ModelLoadModelRequest) msg).getHandler().getBytes(Charset.forName("UTF-8")));
+
+            if (((ModelLoadModelRequest) msg).getGpu() != null) {
+                out.writeInt(Integer.parseInt(((ModelLoadModelRequest) msg).getGpu()));
+            } else {
+                out.writeInt(-1);
+            }
+        } else if (msg instanceof ModelInferenceRequest) {
+            out.writeDouble(1.0);
+            out.writeInt(2); // Predict/inference: 2
+            out.writeInt(msg.getModelName().length());
+            out.writeBytes(msg.getModelName().getBytes(Charset.forName("UTF-8")));
+
+            out.writeInt(-1); // Start of List
+            for (RequestBatch batch : ((ModelInferenceRequest) msg).getRequestBatch()) {
+                encodeRequestBatch(batch, out);
+            }
+            out.writeInt(-2); // End of List
+        }
+        out.writeBytes("\r\n".getBytes(Charset.forName("UTF-8"))); // EOM
+    }
+
+    @Override
+    protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
+        // TODO: Redo this condition. This would work only in current scenario where frontend and backend send and
+        //       receive one request/response at a time.
+
+        if (in.getByte(in.readableBytes() - 1) == '\n') {
+
+            ModelWorkerResponse response = new ModelWorkerResponse();
+
+            response.setCode(Integer.toString(in.readInt()));
```

Comment:
Better make code as int from backend

Reply:
It is int in backend. it is created as String in frontend.